src/Compilers/VisualBasic/Portable/Symbols/PropertySymbol.vb

```diff
@@ -144,13 +144,59 @@ Namespace Microsoft.CodeAnalysis.VisualBasic.Symbols
         ''' <summary>
         ''' Indicates if the property can be written into, which means this 
         ''' property has a setter or it is a getter only autoproperty accessed 
-        ''' in a corresponding constructor or initializer
+        ''' in a corresponding constructor or initializer.
+        ''' If the setter is init-only, we also check that it is accessed in a constructor
+        ''' on Me/MyBase/MyClass or is a target of a member initializer in an object member
+        ''' initializer.
         ''' </summary>
-        Friend Function IsWritable(receiverOpt As BoundExpression, containingBinder As Binder) As Boolean
+        Friend Function IsWritable(receiverOpt As BoundExpression, containingBinder As Binder, isKnownTargetOfObjectMemberInintializer As Boolean) As Boolean
             Debug.Assert(containingBinder IsNot Nothing)
 
-            If Me.HasSet Then
-                Return True
+            Dim mostDerivedSet As MethodSymbol = Me.GetMostDerivedSetMethod()
+
+            If mostDerivedSet IsNot Nothing Then
+                If Not mostDerivedSet.IsInitOnly Then
+                    Return True
+                End If
+
+                If receiverOpt Is Nothing Then
+                    Return False
+                End If
+
+                ' ok: New C() With { .InitOnlyProperty = ... }
```

Comment:
In C# we had to handle nested initializers differently. Does VB also support nested initializers? (same question for `With` expression)

Reply:
> In C# we had to handle nested initializers differently. Does VB also support nested initializers? (same question for With expression)

There is no such thing as a nested initializer in VB. Nested with statements are allowed but only the immediate one is in effect at any given point. In other words, there is no fallback to the outer With statement, the receiver is overwritten. 

---
In reply to: [556982409](https://github.com/dotnet/roslyn/pull/50414#discussion_r556982409) [](ancestors = 556982409)